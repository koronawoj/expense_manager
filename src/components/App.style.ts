import styled, { css } from 'react-emotion';

export const StyleAppWrapper = styled('div')`
  margin: 0 auto;
  max-width: 1200px;
  box-sizing: border-box;
  padding: 20px;
`