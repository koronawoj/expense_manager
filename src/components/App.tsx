import * as React from 'react'
import {Component} from 'react'
import {Provider, observer} from 'mobx-react'
//
import {Expense, ExpensesStore} from '../store/expensesStore'
// import { TodoAdd } from './todo.add'
// import { TodoList } from './todo.list'
@observer
export default class App extends Component {
    private expensesStore: ExpensesStore = new ExpensesStore()

    private handleDelete = (id: number) => {
        this.expensesStore.deleteExpense(id)
    }

    private calculateAmount = (amount:number) => {
        return (amount / 4).toFixed(2)
    }

    render() {
        return (
            <Provider todoStore={this.expensesStore}>
                <div>
                    <div>Expenses: {this.expensesStore.expensesList.length}</div>
                    <button onClick={() => {
                        let date = new Date()
                        this.expensesStore.addExpanse('TitleBig', 123, date.getTime())
                    }}>Big num
                    </button>
                    <button onClick={() => {
                        let date = new Date()
                        this.expensesStore.addExpanse('TitleSmall', 23, date.getTime())
                    }}>Small num
                    </button>
                    <div>List: {this.expensesStore.expensesList.length}</div>
                    <div>Small: {this.expensesStore.expanseUnder100}</div>
                    <div>Big: {this.expensesStore.expensesList.length - this.expensesStore.expanseUnder100}</div>
                    {this.expensesStore.expensesList.map((expense) => (
                        <div key={expense.id}>
                            <span>{expense.title} - {expense.amount} - {this.calculateAmount(expense.amount)}</span>
                            <span onClick={() => {
                            this.handleDelete(expense.id)
                        }}>X</span></div>
                    ))}
                </div>
            </Provider>
        )
    }
}
{/*<Provider todoStore={this.todoStore}>*/
}
{/*<div>*/
}
{/*<TodoAdd />*/
}
{/*<TodoList />*/
}
{/*</div>*/
}
{/*</Provider>*/
}
