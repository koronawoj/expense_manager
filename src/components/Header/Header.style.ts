import styled, { css } from 'react-emotion';

interface IHeader {
    font: string
}

export const StyleHeaderWrapper = styled('div')`
  display: flex;
  justify-content: space-between;
  align-items: center;
`

export const StyleHeaderText = styled('div')<IHeader>`
  font-size: ${props => props.font};
`