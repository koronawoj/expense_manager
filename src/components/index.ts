export * from './Header'
export * from './App'
export * from './AddExpensePanel'
