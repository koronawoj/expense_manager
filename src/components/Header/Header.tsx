import * as React from 'react'
import { StyleHeaderWrapper, StyleHeaderText } from './Header.style'

interface IPropsHeader {
  currentRate: {
    currencyRate: number
    currency: string
  }
}

export class Header extends React.PureComponent<IPropsHeader> {
  public render() {
    return (
      <StyleHeaderWrapper>
        <StyleHeaderText font={'24px'}>List of expenses</StyleHeaderText>
        <StyleHeaderText font={'14px'}>
          1{this.props.currentRate.currency} ={' '}
          {this.props.currentRate.currencyRate} PLN
        </StyleHeaderText>
      </StyleHeaderWrapper>
    )
  }
}
