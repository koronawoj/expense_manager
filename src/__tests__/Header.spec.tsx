import * as React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { Header } from "../components/Header";

const currentRate = {
    currencyRate: 1.2,
    currency: 'PLN'
}

it("renders the heading", () => {
    const result = shallow(<Header currentRate={currentRate}/>);
    expect(toJson(result)).toMatchSnapshot();
});