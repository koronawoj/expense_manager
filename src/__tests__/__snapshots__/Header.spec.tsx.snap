// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the heading 1`] = `
<Styled(div)>
  <Styled(div)
    font="24px"
  >
    List of expenses
  </Styled(div)>
  <Styled(div)
    font="14px"
  >
    1
    PLN
     =
     
    1.2
     PLN
  </Styled(div)>
</Styled(div)>
`;
